packages/discord.js/src/structures/PollBuilder.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+const { PollBuilder: BuildersPoll, normalizeArray } = require('@discordjs/builders');
+const { isJSONEncodable } = require('@discordjs/util');
+const PollAnswerBuilder = require('./PollAnswerBuilder');
+const { toSnakeCase } = require('../util/Transformers');
+const { resolvePartialEmoji } = require('../util/Util');
+
+/**
+ * Represents a poll builder.
+ * @extends {BuildersPoll}
+ */
+class PollBuilder extends BuildersPoll {
+  constructor(data = {}) {
+    super(
+      toSnakeCase({
+        ...data,
+        answers:
+          data.answers &&
+          data.answers.map(answer => {
+            if ('poll_media' in answer) answer = answer.poll_media;
+
+            return {
+              poll_media: {
+                text: answer.text,
+                emoji: answer.emoji && resolvePartialEmoji(answer.emoji),
+              },
+            };
+          }),
+      }),
+    );
+  }
+
+  /**
+   * @typedef {Object} PollAnswerEmojiObject
+   * @property {Snowflake|undfined} id The id of the emoji
+   * @property {string|undefined} name The name of the emoji
+   * @property {boolean|undefined} animated Whether the emoji is animated
+   */
+
+  /**
+   * @typedef {Object} PollAnswerData Data used for an answer on a poll
+   * @property {string} text The text to use for the answer
+   * @property {string|PollAnswerEmojiObject|undefined} emoji The emoji to use for the answer
+   */
+
+  /**
+   * Sets the answers for this poll.
+   * @param {...PollAnswerData} [answers] The answers to set
+   * @returns {PollBuilder}
+   */
+  setAnswers(...answers) {
+    this.spliceAnswers(0, this.data.answers.length, normalizeArray(answers));
+    return this;
+  }
+
+  /**
+   * Appends answers to the poll.
+   * @param {...PollAnswerData} [answers] The answers to add
+   * @returns {PollBuilder}
+   */
+  addAnswers(...answers) {
+    super.addAnswers(
+      normalizeArray(answers).map(answer => {
+        if (typeof answer === 'function') {
+          return answer(new PollAnswerBuilder());
+        }
+        return {
+          text: answer.text,
+          emoji: answer.emoji && resolvePartialEmoji(answer.emoji),
+        };
+      }),
```

Comment:
This logic is duplicated on `spliceAnswers`. We can probably extract it to a separate (private) method